.sti/bin/run

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+function is_puma_installed() {
+  [ ! -f Gemfile.lock ] && return 1
+  grep ' puma ' Gemfile.lock >/dev/null
+}
+
+set -e
+
+function check_number() {
+  if [[ ! "$2" =~ ^[0-9]+$ ]]; then
+    echo "$1 needs to be a non-negative number"
+    exit 1
+  fi
+}
+check_number PUMA_WORKERS     "${PUMA_WORKERS:-0}"
+check_number PUMA_MIN_THREADS "${PUMA_MIN_THREADS:-0}"
+check_number PUMA_MAX_THREADS "${PUMA_MAX_THREADS:-0}"
+
+export RACK_ENV=${RACK_ENV:-"production"}
+
+if bundle exec rake db:migrate >/dev/null; then
+   echo "DB Migrated"
+else
+  echo "ERROR RUNNING DB MIGRATE"
+fi
+
+if is_puma_installed; then
+  export_vars=$(cgroup-limits) ; export $export_vars
+
+  exec bundle exec "puma --config ../etc/puma.cfg"
```

Comment:
Two questions here -
1) Is /etc/puma.cfg a puma default when the gem is installed or should you have a .s2i/bin/assemble script that is copying a cfg file to that location when the image is built
2) should this just use the absolute `/etc` path instead of relative with `../` if ever there was a reason to change the working_directory to something other than the default elsewhere in the script